simple_certmanager/forms.py

```diff
@@ -35,8 +46,12 @@ def __init__(self, *args, **kwargs):
         if not self.instance.pk:
             self.fields["should_renew_csr"].disabled = True
 
-    def save(self, commit=True):
-        instance = super().save(commit=False)
+    def save(self, commit: bool = True) -> Any:
+        instance = super().save(commit)
+        # Handle the certificate file upload
+        validated_certificate = self.cleaned_data.get("certificate")
+        if validated_certificate:
+            instance.create_certificate(validated_certificate)
         if self.cleaned_data.get("should_renew_csr", False):
```

Comment:
Reading this code, it looks like I can now upload a certificate AND generate a new private key/CSR at the same time. That should be prevented:

1. you either generate a new private key/CSR and then no certificate file may be uploaded
2. or you upload a certificate file, and then `should_renew_csr` must be `False`.